ddtrace/contrib/internal/ray/patch.py

```diff
@@ -0,0 +1,439 @@
+    async def _traced_run_method(self: Any, *args: Any, _dd_trace_ctx, **kwargs: Any) -> Any:
+        from ddtrace.ext import SpanTypes
+
+        context = _TraceContext._extract(_dd_trace_ctx)
+        submission_id = os.environ.get(RAY_SUBMISSION_ID)
+
+        with long_running_ray_span(
+            f"{self.__class__.__name__}.{method.__name__}",
+            service=os.environ.get(RAY_JOB_NAME),
+            span_type=SpanTypes.RAY,
+            child_of=context,
+            activate=True,
+        ) as supervisor_run_span:
+            _inject_context_in_env(supervisor_run_span.context)
+
+            try:
+                await method(self, *args, **kwargs)
+            except ray.exceptions.AsyncioActorExit as e:
+                # if the job succeeded we remove from the span
+                # the error used to exit the actor
+                job_info = await self._job_info_client.get_info(submission_id)
+
+                if str(job_info.status) == "FAILED":
+                    raise e
+
+    return _traced_run_method
+
+
+def _exec_entrypoint_wrapper(method: Callable[..., Any]) -> Any:
+    def _traced_exec_entrypoint_method(self: Any, *args: Any, _dd_trace_ctx=None, **kwargs: Any) -> Any:
+        from ddtrace import tracer
+        from ddtrace.ext import SpanTypes
+
+        if tracer.current_span() is None:
+            tracer.context_provider.activate(_extract_tracing_context_from_env())
+
+        script_name_match = ENTRY_POINT_REGEX.search(self._entrypoint)
+        if script_name_match:
+            entrypoint_name = f"{script_name_match.group(1)}.py"
+        else:
+            entrypoint_name = os.path.basename(self._entrypoint)
+
+        with tracer.trace(
+            f"exec {entrypoint_name}", service=os.environ.get(RAY_JOB_NAME), span_type=SpanTypes.RAY
+        ) as span:
+            span.set_tag_str(SPAN_KIND, SpanKind.CONSUMER)
+            _inject_ray_span_tags_and_metrics(span)
+
+            return method(self, *args, **kwargs)
+
+    return _traced_exec_entrypoint_method
+
+
+@contextmanager
+def _trace_actor_method(self: Any, method: Callable[..., Any], dd_trace_ctx, *args, **kwargs):
+    context = tracer.context_provider.active()
+    if context is None:
+        context = _TraceContext._extract(dd_trace_ctx)
+
+    with long_running_ray_span(
+        f"{self.__class__.__name__}.{method.__name__}",
+        service=os.environ.get(RAY_JOB_NAME),
+        span_type=SpanTypes.RAY,
+        child_of=context,
+        activate=True,
+    ) as actor_execute_span:
+        set_maybe_big_tag(actor_execute_span, "ray.actor_method.args", args)
+        set_maybe_big_tag(actor_execute_span, "ray.actor_method.kwargs", kwargs)
+
+        yield actor_execute_span
+
+
+def _inject_tracing_actor_method(method: Callable[..., Any]) -> Any:
+    def _traced_method(self: Any, *args: Any, _dd_trace_ctx=None, **kwargs: Any) -> Any:
+        from ddtrace import tracer
```

Comment:
Do we need to inline this import? This could be a code smell if this is due to a circular import